### ISSUE TITLE
L022 fails on cte with leading comma

### ISSUE THREAD

**Issue by pwildenhain** [sqlfluff/sqlfluff#397]
Using the latest version from `master`

.sqlfluff
```
[sqlfluff:rules]
max_line_length = 115

[sqlfluff:indentation]
indented_joins = True

[sqlfluff:rules:L019]
comma_style = leading
```

test.sql
```sql
with cte_1 as (
    select
        *
    from
        table_1
)

, cte_2 as (
    select
        *
    from
        table_2
)

select * from table_3
```

```
$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   8 | P:   3 | L022 | Blank line expected but not found after CTE definition.
```

If I remove the leading comma config, and make the comma trailing, then it passes

**Comment by alanmcruickshank** [comment]
Good catch, yes on inspection at the moment `L022` doesn't respect `comma_style`, and it definitely should. The `_eval` function for that rule will need to have a path which checks that. Would you like to put together a PR for this?

You would need to do two this:
1. Introduce an `__init__` function for `L022`, almost identical to the one for `L019` which adds the `comma_style` config into the class.
2. Alter `Rule_L022._eval` so that it respects that config value.

**Comment by alanmcruickshank** [comment]
Btw. This is also a duplicate of #351

**Comment by pwildenhain** [comment]
I'm def willing to take a crack at it 🔧

**Comment by pwildenhain** [comment]
Can you assign me to this issue? It will make it easier to track on my end
